src/vs/workbench/browser/parts/views/treeView.ts

```diff
@@ -1183,3 +1214,30 @@ export class CustomTreeView extends TreeView {
 		}
 	}
 }
+
+export class CustomTreeViewDragAndDrop implements ITreeDragAndDrop<ITreeItem> {
+	constructor(private dataSource: TreeDataSource) { }
+
+	onDragOver(data: IDragAndDropData, targetElement: ITreeItem, targetIndex: number, originalEvent: DragEvent): boolean | ITreeDragOverReaction {
+		return { accept: true, bubble: TreeDragOverBubble.Down, autoExpand: true };
+	}
+
+	getDragURI(element: ITreeItem): string | null {
+		return element.tooltip ? element.tooltip.toString() : null;
+	}
+
+	getDragLabel?(elements: ITreeItem[]): string | undefined {
+		if (elements.length > 1) {
+			return String(elements.length);
+		}
+		const element = elements[0];
+		return element.label ? element.label.label : undefined;
```

Comment:
If there's no label, but there is a resource, then we should return the a label for that resource using the label service. Something like:

```ts
this.labelService.getUriLabel(URI.revive(element.resource));
```

You'll then also need to add the label service to your constructor, which you can see other examples of if you search for `labelService` in this file.